compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -403,6 +420,23 @@ convertTypeDef env (ATyCon t)
   | GHC.moduleNameFS (GHC.moduleName (nameModule (getName t))) == "DA.Internal.LF"
   , getOccFS t `elementOfUniqSet` internalTypes
   = pure []
+convertTypeDef env (ATyCon t)
+    -- NOTE(MH): We detect `newtype` definitions produced by the desugring
+    -- of `template instance` declarations and inline the record definition
+    -- of the generic template.
+    | isNewTyCon t
+    , ([], TypeCon tpl args) <- newTyConRhs t
+    , any (\(c, args') -> getOccFS c == getOccFS tpl <> "Instance" && eqTypes args args') $ envInstances env
```

Comment:
Do we really have no better way to detect this than to search for an `Instance` suffix? That seems quite easy to fool.

Reply:
Right now, I can't thing of any better. My plan is to change the class name from `FooInstance` to `Foo:Instance` or something similar that cannot be written by the user since it doesn't match the lexical rules.